phac-nml/irida-next
# Enhancement: Edit line list metadata in place

## What does this PR do and why?
_Describe in detail what your merge request does and why._

This feature allows users to edit metadata directly in line with the project or group samples table.

1. Clicking on a cell in that contains metadata will switch the cell to contain a text input with the value set to the current value.
2. Pressing `escape` while editing will cancel the edit, return to the normal cell with the original value
3. Once the value is changed, clicking outside the cell or pressing enter will commit the value.

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

### Clicking on a metadata cell changes the cell to a text field
![image](https://github.com/user-attachments/assets/f6c0a81e-c483-4632-9e65-044e67b9bc63)

### Error if clicking on a metadata field that the user is not allowed to edit
![image](https://github.com/user-attachments/assets/d82ab6de-ba3b-4317-848a-3e04bc1fd66b)

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1. Login as user5
4. Go to a project that the user has permissions for and go to the samples page
5. Toggle on metadata
6. Click on any cell in the table that has metadata and edit a field. 
7. Go to http://localhost:3000/-/groups/streptococcus/-/samples?limit=20&page=1 (User cannot edit here)
8. Attempt to edit field, error should be displayed

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
